### DISCUSSION TITLE
IClassFixture with WebApplicationFactory and one time test setup

### DISCUSSION THREAD

**Discussion by dmccolloughOneGas** [xunit/xunit#2865] [Question] [answered]
I have some data that I would like to fetch once when the test class is created. I've read some of the docs but I'm struggling how to do this correctly in my situation.

The docs say that I should create a class fixture and implement IDisposable, but how do I do that and also use WebApplicationFactory ?

I want to use WebApplicationFactory so that I can access some settings from the appsettings.json file in my tests.

my test class looks like this
```
public class SomeControllerTests : IClassFixture<WebApplicationFactory<Program>>
{
   public SomeControllerTests(WebApplicationFactory<Program> factory)
   {
      // Fetch some data ONCE to be shared between each of the tests.
      // The data I want to share is the result of an HTTP request.
   }

   [Fact]
   public void SomeTest()
   {
       // Use fetched data from the constructor.
   }
}
```

**Comment by bartelink** [comment, marked as answer]
https://xunit.net/docs/shared-context shows how you write a fixture (the init logic lives in the fixture, not where you have the logic in the ctor)

**Reply by dmccolloughOneGas** [reply]
Thanks Ruben, I'll take a look at that.

**Reply by dmccolloughOneGas** [reply]
Thanks Ruben, this is what I was needing. Much appreciated.
